openaerostruct/functionals/moment_coefficient.py

```diff
@@ -174,34 +183,73 @@ def compute_partials(self, inputs, partials):
+
+            # Compute the derivative wrt to the first element of diff
             dcda[0, :, :, 1] = sec_forces[:, :, 2]
             dcda[0, :, :, 2] = -sec_forces[:, :, 1]
+
+            # Compute the derivative wrt to the second element of diff
             dcda[1, :, :, 0] = -sec_forces[:, :, 2]
             dcda[1, :, :, 2] = sec_forces[:, :, 0]
+
+            # Compute the derivative wrt to the third element of diff
             dcda[2, :, :, 0] = sec_forces[:, :, 1]
             dcda[2, :, :, 1] = -sec_forces[:, :, 0]
 
+            # Compute the derviative of the moment vectors(c) with respect to the sec_forces vectors(b) multiplied by -1
             dcdb = np.zeros((3, nx - 1, ny - 1, 3))
+
+            # Compute the derivative wrt to the first element of sec_forces
             dcdb[0, :, :, 1] = -diff[:, :, 2]
             dcdb[0, :, :, 2] = diff[:, :, 1]
+
+            # Compute the derivative wrt to the second element of sec_forces
             dcdb[1, :, :, 0] = diff[:, :, 2]
             dcdb[1, :, :, 2] = -diff[:, :, 0]
+
+            # Compute the derivative wrt to the third element of sec_forces
             dcdb[2, :, :, 0] = -diff[:, :, 1]
             dcdb[2, :, :, 1] = diff[:, :, 0]
 
+            # Compute derivative of CM wrt to the sec_forces of the section by reshaping to 3 rows and multiplying by fact.
             partials["CM", name + "_sec_forces"] += dcdb.reshape((3, 3 * (nx - 1) * (ny - 1))) * fact
 
+            # Compute derivative of M wrt to the sec_forces of the section by reshaping to 3 rows and multiplying by fact.
+            partials["M", name + "_sec_forces"] += dcdb.reshape((3, 3 * (nx - 1) * (ny - 1)))
+
+            # Project the derviative of the moment vectors(c) with respect to the diff vectors(a)
+            # onto the derivative of mid-panel chord distribution wrt to the chord distribution giving the derivative of
+            # the moment vectors(c) with respect to the chord distribution(dc_dchord). This works because the diff
+            # vectors are difference between the mid-panel bound vortex(quarter chord) points and cg which is static in this derivative.
+            # The spanwise component of the mid-panel bound vortex(quarter chord) points have the same derivatrive wrt to the chord
+            # distribution as the mid-panel chord distribution does wrt the the chord distribution.
             dc_dchord = np.einsum("ijkl,km->ijml", dcda, dpc_dc)
+
+            # Compute the derivative of CM wrt to the bound vortex points(b_pts) by reshaping dc_dchord to three rows
+            # and multiplying by fact.
             partials["CM", name + "_b_pts"] += dc_dchord.reshape((3, 3 * (nx - 1) * ny)) * fact
 
+            # Compute the derivative of M wrt to the bound vortex points(b_pts) by reshaping dc_dchord to three rows
+            # and multiplying by fact.
+            partials["M", name + "_b_pts"] += dc_dchord.reshape((3, 3 * (nx - 1) * ny))
```

Comment:
I don't see a multiplication by `fact` as you mention in the comments

Reply:
Error in the comment. fact is just the (1/q*S*MAC) term which normalizes to CM so we get rid of it when we do the moment derivative.